app/src/main/java/com/android/sample/ui/activity/ActivityForm.kt

```diff
@@ -0,0 +1,400 @@
+  if (locationSuggestions.size > 3) {
+    DropdownMenuItem(
+        text = { Text("More...") },
+        onClick = { /* TODO: Define behavior for 'More...' */},
+        modifier = Modifier.padding(STANDARD_PADDING.dp))
+  }
+
+  Spacer(modifier = Modifier.height(STANDARD_PADDING.dp))
+
+  ExposedDropdownMenuBox(
+      modifier = Modifier.testTag("chooseTypeMenu").fillMaxWidth().padding(STANDARD_PADDING.dp),
+      expanded = expandedType,
+      onExpandedChange = onExpandedTypeChange) {
+        OutlinedTextField(
+            readOnly = true,
+            value = selectedOptionType,
+            onValueChange = {},
+            label = { Text(context.getString(R.string.activity_type)) },
+            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = expandedType) },
+            colors = ExposedDropdownMenuDefaults.textFieldColors(),
+            modifier = Modifier.menuAnchor().fillMaxWidth().testTag("typeTextField"))
+        ExposedDropdownMenu(
+            expanded = expandedType,
+            onDismissRequest = onDismissType,
+            modifier = Modifier.fillMaxWidth().padding(STANDARD_PADDING.dp)) {
+              types.forEach { selectionOption ->
+                DropdownMenuItem(
+                    modifier = Modifier.fillMaxWidth().padding(STANDARD_PADDING.dp),
+                    text = { Text(selectionOption.name) },
+                    onClick = { onSelectType(selectionOption) })
+              }
+            }
+      }
+
+  ExposedDropdownMenuBox(
+      modifier = Modifier.testTag("chooseCategoryMenu").fillMaxWidth().padding(STANDARD_PADDING.dp),
+      expanded = expandedCategory,
+      onExpandedChange = onExpandedCategoryChange) {
+        OutlinedTextField(
+            readOnly = true,
+            value =
+                selectedOptionCategory?.name
+                    ?: context.getString(R.string.select_activity_category),
+            onValueChange = {},
+            label = { Text(context.getString(R.string.activity_category)) },
+            trailingIcon = {
+              ExposedDropdownMenuDefaults.TrailingIcon(expanded = expandedCategory)
+            },
+            colors = ExposedDropdownMenuDefaults.textFieldColors(),
+            modifier = Modifier.menuAnchor().fillMaxWidth().testTag("categoryTextField"))
+        ExposedDropdownMenu(
+            expanded = expandedCategory,
+            onDismissRequest = onDismissCategory,
+            modifier = Modifier.fillMaxWidth().padding(STANDARD_PADDING.dp)) {
+              interestsCategories.forEach {
+                DropdownMenuItem(
+                    modifier = Modifier.fillMaxWidth().padding(STANDARD_PADDING.dp),
+                    text = { Text(it.name) },
+                    onClick = { onSelectCategory(it) })
+              }
+            }
+      }
+
+  if (selectedOptionCategory != null) {
+    Spacer(modifier = Modifier.height(SMALL_PADDING.dp))
+
+    ExposedDropdownMenuBox(
```

Comment:
These 3 exposedDropDownMenuBoxes are very similar, first you can create a customized one so that you use it everywhere and make testing way simpler. Also this will reduce code duplication

Reply:
Totally aggree with it, working on it as soon as possible